File: frontend/tattool-ts/src/model/transport/execution/PayloadExecutionEvent.ts

enum Status {
    SUCCESS,
    FAILURE,
    RUNNING
}


interface PayloadExecutionEvent {

    readonly platformId: string;

    readonly recordingId: string;

    readonly recordingDisplayName: string;

    readonly recordingInstanceId: string;

    readonly id: string;

    readonly displayName: string;

    readonly payload: any;

    readonly status: Status;

    readonly durationInMs: number;

    readonly datasetRow: Array<string>;
}

interface RunTerminatedExecutionEvent {
    readonly runId: string;
}




export { PayloadExecutionEvent, RunTerminatedExecutionEvent, Status }

File: backend/core/src/main/java/ch/qarts/tattool/core/domain/execution/live/PayloadExecutionEvent.java
package ch.qarts.tattool.core.domain.execution.live;

import ch.qarts.tattool.core.domain.execution.Status;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PayloadExecutionEvent implements ExecutionEvent {

    private String platformId;

    private String recordingId;

    private String recordingDisplayName;

    private String recordingInstanceId;

    private String id;

    private String displayName;

    private Object payload;

    private Status status;

    private Long durationInMs;

    private List<String> datasetRow;

}